stable/prometheus-operator/templates/alertmanager/alertmanager.yaml

```diff
@@ -55,8 +55,12 @@ spec:
   podMetadata:
 {{ toYaml .Values.alertmanager.alertmanagerSpec.podMetadata | indent 4 }}
 {{- end }}
-{{- if eq .Values.alertmanager.alertmanagerSpec.podAntiAffinity "hard" }}
+{{- if or .Values.alertmanager.alertmanagerSpec.podAntiAffinity .Values.alertmanager.alertmanagerSpec.affinity }}
   affinity:
+{{- if .Values.alertmanager.alertmanagerSpec.affinity }}
+{{ toYaml .Values.alertmanager.alertmanagerSpec.affinity | indent 4 }}
+{{- end }}
+{{- if eq .Values.alertmanager.alertmanagerSpec.podAntiAffinity "hard" }}
```

Comment:
would it make sense to check if `.Values.alertmanager.alertmanagerSpec.affinity.podAntiAffinity` is defined?

Reply:
It's being implicitly checked, because there are two ifs for `hard` and `soft` but no else statement. If it is not defined (the default) then the template would not output anything for `podAntiAffinity`.